Kernel/Devices/PATADiskDevice.h

```diff
@@ -39,37 +37,28 @@ class IDEDiskDevice final : public IRQHandler
     virtual bool read_blocks(unsigned index, u16 count, u8*) override;
     virtual bool write_blocks(unsigned index, u16 count, const u8*) override;
 
+    void set_drive_geometry(u16, u16, u16);
+
 protected:
-    explicit IDEDiskDevice(DriveType);
+    explicit PATADiskDevice(PATAChannel&, DriveType);
 
 private:
-    // ^IRQHandler
-    virtual void handle_irq() override;
 
     // ^DiskDevice
     virtual const char* class_name() const override;
 
-    void initialize();
     bool wait_for_irq();
     bool read_sectors_with_dma(u32 lba, u16 count, u8*);
     bool write_sectors_with_dma(u32 lba, u16 count, const u8*);
     bool read_sectors(u32 lba, u16 count, u8* buffer);
     bool write_sectors(u32 lba, u16 count, const u8* data);
-
     bool is_slave() const;
 
     Lock m_lock { "IDEDiskDevice" };
     u16 m_cylinders { 0 };
     u16 m_heads { 0 };
     u16 m_sectors_per_track { 0 };
-    u16 m_io_base { 0 };
-    volatile bool m_interrupted { false };
-    volatile u8 m_device_error { 0 };
+    DriveType m_drive_type { DriveType::Master };
 
-    DriveType m_drive_type { DriveType::MASTER };
-    PCI::Address m_pci_address;
-    PhysicalRegionDescriptor m_prdt;
-    RefPtr<PhysicalPage> m_dma_buffer_page;
-    u16 m_bus_master_base { 0 };
-    Lockable<bool> m_dma_enabled;
+    PATAChannel& m_ptr_channel;
```

Comment:
@awesomekling I'm not too sure if this is okay. I was struggling a bit with `AK` so I felt it was best to just use a normal reference.

Reply:
This looks correct.
A `PATADiskDevice` object is always owned by a `PATAChannel` object, and so we know that the `PATAChannel` will always exist as long as the `PATADiskDevice` does.
When we know that something is guaranteed to exist, and we're not responsible for keeping it alive, a plain reference is the most appropriate. :)
I'd suggest dropping `ptr` from the name though, since it's not a pointer.

Reply:
Ahh good, I wasn't 100% sure but I assumed as much because the disk exists as long as it's actually connected to the channel.

I'll change the name now :)